policy/scan/error_reporter.go

```diff
@@ -0,0 +1,90 @@
+package scan
+
+import (
+	"strings"
+
+	"go.mondoo.com/cnquery/cli/theme"
+	"go.mondoo.com/cnquery/motor/asset"
+	"go.mondoo.com/cnspec/policy"
+	pbStatus "go.mondoo.com/ranger-rpc/status"
+	"google.golang.org/genproto/googleapis/rpc/errdetails"
+)
+
+type ErrorReporter struct {
+	assets map[string]*policy.Asset
+	errors map[string]string
+}
+
+func NewErrorReporter(assetList []*asset.Asset) Reporter {
+	assets := make(map[string]*policy.Asset, len(assetList))
+	for i := range assetList {
+		cur := assetList[i]
+		assets[cur.Mrn] = &policy.Asset{
+			Mrn:  cur.Mrn,
+			Name: cur.Name,
+			Url:  cur.Url,
+		}
+	}
+	return &ErrorReporter{assets: assets, errors: make(map[string]string)}
+}
+
+func (r *ErrorReporter) AddReport(asset *asset.Asset, results *AssetReport) {}
+
+func (c *ErrorReporter) AddScanError(assetObj *asset.Asset, err error) {
+	if c.errors == nil {
+		c.errors = make(map[string]string)
+	}
+	name := findNameForAsset(assetObj)
```

Comment:
The other reports do not try to find the name, they assume the mrn is present. Is this intended?

Reply:
This is definitely changing after #172 is merged. That logic has changed there. I am waiting on it to be approved before I refactor this bit. For now it works but it will need to change

Reply:
This function actually just gets the name which is either the name or the mrn for the asset. That is used just for printing the error. I think the code is still correct.